doc/contribute/guidelines.rst

```diff
@@ -536,12 +536,28 @@ reference manuals, etc.
 Coding Style
 ============
 
+.. note::
+   Coding style is enforced on any new or modified code, but contributors are
+   not expected to correct the style on existing code that they are not
+   modifying.
+
+.. note::
+   For style aspects where the guidelines don't offer explicit guidance or
+   permit multiple valid ways to express something, contributors should follow
+   the style of existing code in the tree, with higher importance given to
+   "nearby" code (first look at the function, then the same file, then
+   subsystem, etc).
+
 .. _Linux kernel coding style:
    https://kernel.org/doc/html/latest/process/coding-style.html
 
+.. _snake case:
+   https://en.wikipedia.org/wiki/Snake_case
+
 In general, follow the `Linux kernel coding style`_, with the following
-exceptions:
+exceptions and clarifications:
 
+* Use `snake case`_ for code and variables.
```

Comment:
but this is covered by linux already, in details, I do not think we need to repeat this here:

https://kernel.org/doc/html/latest/process/coding-style.html#naming

Reply:
It's not explicit some said, so it was made explicit here.

Reply:
I'm in favor of incremental additions to our style guide that eventually make reference the Linux style guide obsolete.  While unlikely, Linux can change their style guidelines at any time.

Reply:
need to just fork the original linux style into our docs, call them zephyr style guidelines and never look back again.

Reply:
> kernel.org/doc/html/latest/process/coding-style.html#naming

the more I look at this, the more I think we should just go our own and stop referencing this from Linux. It looks more of a rant than guidelines in some parts.

Reply:
> > kernel.org/doc/html/latest/process/coding-style.html#naming
> 
> the more I look at this, the more I think we should just go our own and stop referencing this from Linux. It looks more of a rant than guidelines in some parts.

Yes, totally, and I wanted to point this out the other day. It seriously feels *very* dated in the way it's written, to the point that it's not even reflecting the values we try to promote via our Code of Conduct, for example... Like, doing something wrong being refered to as being a "shooting offense"... Really?! So yeah if there's something where we can clearly state "we're not Linux" maybe it can start with coding style :) big +1 from me

Reply:
Those are very good arguments. I've never before read the Linux coding guidelines like that, but you are absolutely right.